src/native/windows/src/rprocess.c

```diff
@@ -316,13 +373,16 @@ static BOOL __apxProcessClose(APXHANDLE hProcess)
     lpProc = APXHANDLE_DATA(hProcess);
     CHECK_IF_ACTIVE(lpProc);
 
+    __apxProcessTerminateChild(lpProc->stProcInfo.dwProcessId, TRUE);
     /* Try to close the child's stdin first */
```

Comment:
Can you explain the purpose of the above call? It seems odd to be forcibly terminating child processes this early. What am I missing?

Reply:
Mixing TRUE and FALSE probably...
https://github.com/jfclere/commons-daemon/blob/master/src/native/windows/src/rprocess.c#L413
Here the process MUST be terminated not in line 376.
Good catch I will fix and retest on Monday.

Reply:
Got it. It was `dryrun` being `TRUE` that I missed. This makes sense now. It generates a debug log of the current child processes before a normal shutdown. The forced shutdown only happens if the normal one goes wrong. LGTM.

Reply:
Cool should I add a comment in the code to explain it traces the process tree and another where it kills the process tree?

Reply:
It was obvious once you explained it but a comment wouldn't hurt.

Reply:
Done  Many thanks for the review.

Reply:
> Cool should I add a comment in the code to explain it traces the process tree and another where it kills the process tree?

👍 for comments.